setuptools_rust/setuptools_ext.py

```diff
@@ -48,15 +48,27 @@ def initialize_options(self) -> None:
         def make_distribution(self) -> None:
             if self.vendor_crates:
                 manifest_paths = []
+
+                # Collate cargo manifest options together.
+                # We can cheat here, as the only valid options are the simple strings
+                # --frozen, --locked, or --offline.
+                #
+                # https://doc.rust-lang.org/cargo/commands/cargo-build.html#manifest-options
+                cargo_manifest_args: Set[str] = set()
                 for ext in self.distribution.rust_extensions:
                     manifest_paths.append(ext.path)
+                    if ext.cargo_manifest_args:
+                        cargo_manifest_args.union(ext.cargo_manifest_args)
```

Comment:
I think this call to `union` doesn't actually update the set?

```python
>>> x = set()
>>> x.union({1, 2})
{1, 2}
>>> x
set()
```

I think you want

```suggestion
                        cargo_manifest_args |= set(ext.cargo_manifest_args)
```

Reply:
:facepalm: I blame it on my currently-full-of-covid brain.

Updated to an `update` call, which does a mutable update in place.